File: src/main/java/no/nav/bidrag/beregn/saertilskudd/rest/exception/SjablonConsumerException.java
package no.nav.bidrag.beregn.saertilskudd.rest.exception;

import org.springframework.http.HttpStatus;
import org.springframework.web.client.RestClientResponseException;

public class SjablonConsumerException extends RuntimeException {

  private final HttpStatus statusCode;

  public HttpStatus getStatusCode() {
    return statusCode;
  }

  public SjablonConsumerException(RestClientResponseException exception) {
    super(exception);
    this.statusCode = HttpStatus.valueOf(exception.getRawStatusCode());
  }
}

File: src/main/kotlin/no/nav/bidrag/beregn/saertilskudd/rest/exception/SjablonConsumerException.kt
package no.nav.bidrag.beregn.saertilskudd.rest.exception

import org.springframework.http.HttpStatus
import org.springframework.web.client.RestClientResponseException

class SjablonConsumerException(exception: RestClientResponseException) : RuntimeException(exception) {
    val statusCode: HttpStatus

    init {
        statusCode = HttpStatus.valueOf(exception.rawStatusCode)
    }
}
